### PR TITLE
Extend the description of the TpetraWrappers classes.

### PR DESCRIPTION
**Pull request by kinnewig**
I've added a description of the difference between the TrilinosWrappers and the TpetraWrappers.

I don't know if the last paragraph fits into this section. Please let me know if you think it is inappropriate so I can remove it.

FYI @jpthiele

### CODE REVIEW

File: 9.6/paper.tex [hunk 1]
```diff
@@ -470,12 +470,41 @@ \subsection{Interface to the \trilinos{} \tpetra stack}\label{sec:tpetra}
...
+    be provided at creating the vector object.
+\end{itemize}
 
-All the \texttt{TpetraWrappers} classes take \texttt{Number} and \texttt{MemorySpace} template arguments
-allowing to control the underlying number type and the memory space used for storing the data. Note that the memory space type
-also determines whether \tpetra{} takes advantage of GPUs if \texttt{Kokkos} has been configure with a GPU backend.
+One other important change is that
+all the \texttt{TpetraWrappers} classes take \texttt{Number} and \texttt{MemorySpace} template arguments,
+which control the underlying number type and memory space. 
+By default, the number type is set to  \texttt{double}, and memory is allocated on the CPU, mirroring the
```
Comment (masterleinad) [inline 9.6/paper.tex:495]: There is no default. The `Number` template parameter has to be provided.
```suggestion
By default, the memory is allocated on the CPU, mirroring the 
```

File: 9.6/paper.tex [hunk 2]
```diff
@@ -470,12 +470,41 @@ \subsection{Interface to the \trilinos{} \tpetra stack}\label{sec:tpetra}
...
+By default, the number type is set to  \texttt{double}, and memory is allocated on the CPU, mirroring the 
+behavior of the \texttt{TrilinosWrappers}. 
+The support for different number types also enables the usage of automatic differentiation-based tools in 
+\trilinos{}, such as \texttt{NOX}. 
+Additionally, the memory space type determines whether \tpetra{} can utilize GPUs if \kokkos{} is configured 
+with a GPU backend. 
+To make the automatic differentiation tools of \trilinos{} accessible through the \tpetra{} interface without 
+requiring users to have an in-depth understanding of \trilinos{}, additional wrappers for the corresponding 
+\trilinos{} packages will be added in future releases. 
+Furthermore, further work is needed to fully leverage the GPU backend and unlock its full potential for users.
```
Comment (masterleinad) [inline 9.6/paper.tex:504]: What do you mean by that?
```suggestion
Furthermore, more work is needed to fully leverage the GPU backend and unlock its full potential for users.
```

File: 9.6/paper.tex [hunk 3]
```diff
@@ -470,12 +470,41 @@ \subsection{Interface to the \trilinos{} \tpetra stack}\label{sec:tpetra}
...
+which control the underlying number type and memory space. 
+By default, the number type is set to  \texttt{double}, and memory is allocated on the CPU, mirroring the 
+behavior of the \texttt{TrilinosWrappers}. 
+The support for different number types also enables the usage of automatic differentiation-based tools in 
+\trilinos{}, such as \texttt{NOX}. 
+Additionally, the memory space type determines whether \tpetra{} can utilize GPUs if \kokkos{} is configured 
+with a GPU backend. 
+To make the automatic differentiation tools of \trilinos{} accessible through the \tpetra{} interface without 
+requiring users to have an in-depth understanding of \trilinos{}, additional wrappers for the corresponding 
+\trilinos{} packages will be added in future releases.
```
Comment (masterleinad) [inline 9.6/paper.tex:503]: I wouldn't promise anything about future work we will do.

File: 9.6/paper.tex [hunk 4]
```diff
@@ -470,12 +470,41 @@ \subsection{Interface to the \trilinos{} \tpetra stack}\label{sec:tpetra}
...
+\trilinos{}, such as \texttt{NOX}. 
+Additionally, the memory space type determines whether \tpetra{} can utilize GPUs if \kokkos{} is configured 
+with a GPU backend. 
+To make the automatic differentiation tools of \trilinos{} accessible through the \tpetra{} interface without 
+requiring users to have an in-depth understanding of \trilinos{}, additional wrappers for the corresponding 
+\trilinos{} packages will be added in future releases. 
+Furthermore, further work is needed to fully leverage the GPU backend and unlock its full potential for users.
+
+Moreover, the \tpetra{} interface allows access to more functionality and preconditioners, e.g., \texttt{FROSch}, 
+available in \trilinos{}, which will be made accessible through the \texttt{TpetraWrappers} in the future.
```
Comment (masterleinad) [inline 9.6/paper.tex:507]: Same here.

File: 9.6/paper.tex [hunk 5]
```diff
@@ -470,12 +470,41 @@ \subsection{Interface to the \trilinos{} \tpetra stack}\label{sec:tpetra}
...
-Most of the interfaces are usable at this point even if some functionality existing in the \texttt{TrilinosWrappers} classes
-is still missing, most noticably a wrapper for an algebraic-multigrid preconditioner.
+The preconditioner available in the \texttt{TpetraWrappers::Precondition*} classes can be used along with the existing 
+iterative solvers, e.g., CG, GMRES, etc.; therefore, the interface is already usable. 
+Even if some functionality exists, the \texttt{TrilinosWrappers} classes are missing, most noticably a wrapper for an 
+algebraic-multigrid preconditioner. 
+
+The design goal was to introduce as few changes as possible for the user to allow for an easy transition to the 
+\texttt{TpetraWrappers} classes. 
+However, based on the difference between \epetra{} and \tpetra{}, the following changes where necessary:
```
Comment (jpthiele) [inline 9.6/paper.tex:480]:
```suggestion
However, based on the difference between \epetra{} and \tpetra{}, the following changes were necessary:
```

File: 9.6/paper.tex (hunk 3 above)
Comment (bangerth) [reply 9.6/paper.tex:503]: But you can say "...will require further work". I think that's neutral enough.

File: 9.6/paper.tex (hunk 2 above)
Comment (bangerth) [reply 9.6/paper.tex:504]: Let's not have "Furthermore, more..." :-)

Comment (kinnewig) [reply 9.6/paper.tex:504]: I replaced the "further" with "additional"

**Comment by peterrum** [comment]
Let's merge as is. We can improve the text in the next days :+1:

**Pull request merged**